### DISCUSSION TITLE
Rotary encoder response very slow

### DISCUSSION THREAD

**Discussion by ChKK1963** [probonopd/MiniDexed#253] [Q&A]
Finally connected my display (i2s) and rotary encoder today.

Display was working great immediately with the default 0x27 address.

Encoder is working, too. But response time even for very simple actions feels VERY slow. It does not look like any encoder "clicks" are missed, it is sufficient to turn the knob one click left/right to get a response. But for example, when in the "Reverb/Enable" menu, switching between on and off repeatedly takes 4-5 seconds. Same for all other actions I have checked like navigating up and down in the menu tree etc.

Looks like multiple clicks in a row are processed with the same unexpected initial delay, but then much faster - for example when I am on TG1 and turn 3 the encoder clicks to the right, the display switches to TG2 after 4-5 seconds, but then from TG2 to TG3 and then TG3 to TG4 only with a very short (maybe 300ms) delay.

Any suggestions?

This is using the MiniDexed_2022-05-21-2f044b6 build on a Raspberry Pi 4 B.

**Comment by Banana71** [comment]
this should solve your problems
https://github.com/probonopd/MiniDexed/discussions/178#discussion-4035975

**Comment by sou7611** [comment]
I installed the caps, as noted in #178, and the slow response I saw has gone away.

Now it only registers every second click of the knob.  Button presses work fine.

**Comment by BobanSpasic** [comment]
@sou7611 
did you re-check the values of your resistors and capacitors? Wrong values can get some impulses lost.
Also, do not use tantalum or electrolytic capacitors (with positive and negative leads). You need ceramic or equivalent capacitors.

**Comment by sou7611** [comment]
Yes, ceramic 10n capacitors.  It's very repeatable, only every other click
registers, no matter how fast or slow I spin the knob.  and yes, values are
correct.

It's sending the pulses, I verified that with a logic probe.

**Reply by probonopd** [reply]
I think there are different types of rotary encoders. Maybe yours uses a slightly different encoding scheme?

**Comment by sou7611** [comment]
It's a KY-040 encoder as specified ...

**Reply by probonopd** [reply]
Possibly related to https://github.com/probonopd/MiniDexed/issues/89 ("Different rotary encoders may complete different fractions of a Gray-code cycle per detent"); not sure though.
